### PR TITLE
sharding: WIP collator withdraw

### PR DESCRIPTION
**Pull request by terenc3t**
Adding methods and tests to let collator deregister from SMC's collator pool, and withdraw (release) it's deposit from collator pool

This is dependent on #78 and will not be merged until #78 is throughly tested and merged

### CODE REVIEW

File: sharding/collator/collator.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func checkSMCForCollator(c client.Client, head *types.Header) error {
 	period := big.NewInt(0).Div(head.Number, big.NewInt(sharding.PeriodLength))
 	for s := int64(0); s < sharding.ShardCount; s++ {
 		// Checks if we are an eligible collator according to the SMC
-		addr, err := c.SMCCaller().GetEligibleCollator(&bind.CallOpts{}, big.NewInt(s), period)
+		addr, err := c.SMCCaller().Get_eligible_collator(&bind.CallOpts{}, big.NewInt(s), period)
```
Comment (prestonvanloon) [inline sharding/collator/collator.go]: How did this happen? Please change it back

Comment (terenc3t) [reply sharding/collator/collator.go]: new method name from SMC (phase 1) bindings #78

Comment (prestonvanloon) [reply sharding/collator/collator.go]: ok I'll bring it up with @enriquefynn

**Comment by rauljordan** [comment]
Is this still required based on what's going on in #93 @terenc3t ?

**Comment by terenc3t** [comment]
@rauljordan It is. This PR wraps around the methods defined in #93. We can close this after #93

**Pull request closed without merging**